src/MessagingClient.ts

```diff
@@ -284,6 +290,76 @@ export class MessagingClient implements MessagingAPI {
         return conversation
     }
 
+    /** Get or create a channel with the given users
+     * If the channel already exists this will return the channel and won't invite the passed ids
+     * If the channel is created, all user ids will be invited to join
+     */
+    async createChannel(channelName: string, userIds: SocialId[]): Promise<Conversation> {
+        try {
+            const { room_id } = await this.matrixClient.createRoom({
```

Comment:
Does it make sense to reuse the `getOrCreateConversation` method here instead?

Reply:
Also, thinking if we should validate that the channel name is not a DM name instead.

Reply:
removed this method and validate channel name with a regex